ntoskrnl/ps/query.c

```diff
@@ -186,36 +179,49 @@ NtQueryInformationProcess(
             /* Indicate success */
             Status = STATUS_SUCCESS;
 
-            _SEH2_TRY
+            RtlZeroMemory(&QuotaLimits, sizeof(QuotaLimits));
+
+            /* Get max/min working set sizes */
+            QuotaLimits.MaximumWorkingSetSize =
+                Process->Vm.MaximumWorkingSetSize << PAGE_SHIFT;
+            QuotaLimits.MinimumWorkingSetSize =
+                Process->Vm.MinimumWorkingSetSize << PAGE_SHIFT;
+
+            /* Get default time limits */
+            QuotaLimits.TimeLimit.QuadPart = -1LL;
+
+            /* Is quota block a default one? */
+            if (Process->QuotaBlock == &PspDefaultQuotaBlock)
             {
-                /* Set max/min working set sizes */
-                QuotaLimits->MaximumWorkingSetSize =
-                        Process->Vm.MaximumWorkingSetSize << PAGE_SHIFT;
-                QuotaLimits->MinimumWorkingSetSize =
-                        Process->Vm.MinimumWorkingSetSize << PAGE_SHIFT;
+                /* Get default pools and pagefile limits */
+                QuotaLimits.PagedPoolLimit = (SIZE_T)-1;
+                QuotaLimits.NonPagedPoolLimit = (SIZE_T)-1;
+                QuotaLimits.PagefileLimit = (SIZE_T)-1;
+            }
+            else
+            {
+                /* Get limits from non-default quota block */
+                QuotaLimits.PagedPoolLimit =
+                    Process->QuotaBlock->QuotaEntry[PsPagedPool].Limit;
+                QuotaLimits.NonPagedPoolLimit =
+                    Process->QuotaBlock->QuotaEntry[PsNonPagedPool].Limit;
+                QuotaLimits.PagefileLimit =
+                    Process->QuotaBlock->QuotaEntry[PsPageFile].Limit;
+            }
 
-                /* Set default time limits */
-                QuotaLimits->TimeLimit.LowPart = MAXULONG;
-                QuotaLimits->TimeLimit.HighPart = MAXULONG;
+            /* Get additional information, if needed */
+            if (Extended)
+            {
+                /* FIXME: Get the correct information */
+                //QuotaLimits.WorkingSetLimit = (SIZE_T)-1; // Not used on Win2k3, it is set to 0
```

Comment:
I'm not sure if I'm reading this correctly but what is considered as questioned regarding information correctness? The `WorkingSetLimit`?

Reply:
The old versions of this struct label this field as `Reserved1` and it's set to 0 on win2k3 so I assume it's reserved (not used).
[https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/include/xdk/winnt_old.h;h=157c3724e8096583045bd8b2cb1a20445e9bf1d1;hb=c10c5224ff76aeda1d83273e736a8a6244d74f4e#l2537](https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/include/xdk/winnt_old.h;h=157c3724e8096583045bd8b2cb1a20445e9bf1d1;hb=c10c5224ff76aeda1d83273e736a8a6244d74f4e#l2537)

Wine uses `(SIZE_T)-1` for it and our code doesn't actually support `QUOTA_LIMITS_EX` internally so I use the Wine's values.
[https://gitlab.winehq.org/wine/wine/-/merge_requests/5507/diffs#a666ff29c09175a075770ae07b2d8ca32ee2f5e5_1602_1604](https://gitlab.winehq.org/wine/wine/-/merge_requests/5507/diffs#a666ff29c09175a075770ae07b2d8ca32ee2f5e5_1602_1604)